### ISSUE TITLE
False positives of UnusedVariable in fn() functions

### ISSUE THREAD

**Issue by ghnp5** [sirbrillig/phpcs-variable-analysis#346]
Hello,

Since the last version (or some earlier, but recently), I keep getting false positives of `VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable` in arrow functions.

Example:

```
$ignoredDirs = array_map(fn($dir) => PREFIX_DIR.$dir, PHP_ERROR_LOGS['ignore_dirs']);
```

It says that `$dir` is unused.

It happens for all arrow functions, seems, and for all parameters of those functions.

Using PHP 8.3, on Windows.

Thanks!

**Comment by sirbrillig** [comment]
Thanks for the report! The scope for functions is tricky to get right and I definitely could have broken something. 

I'm not able to reproduce the issue in my test classes, though. 🤔 I wonder what's causing it. Here's the test case I added. If you put this into your code, does it report a warning?

```php
function arrowFunctionWithConcatOperator() {
	$ignoredDirs = array_map(fn($dir) => PREFIX_DIR.$dir, PHP_ERROR_LOGS['ignore_dirs']);
	return $ignoredDirs;
}
```

If not, maybe it's interacting with the other code around the line you posted above. Can you come up with a small function like the above that demonstrates the false positive?

**Comment by sirbrillig** [comment]
Ah wait! I see, I get the false positive if I put that line at the top level of a file.

```php
array_map(
	fn($dir) => PREFIX_DIR.$dir, // false positive warning
	PHP_ERROR_LOGS['ignore_dirs']
);
```

I probably didn't account for arrow functions outside of a function scope.

**Comment by ghnp5** [comment]
Thanks! Surprisingly, that doesn't trigger the warning. But if you put the line of code in the global scope (outside any function or class), that will trigger the warning.

EDIT - I see you already figured that part out 😁

**Comment by sirbrillig** [comment]
Should be fixed in 2.11.22. Thanks for reporting!
